### PR TITLE
Installer leftovers

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Implement two TODOs from #8576 
Changes proposed in this pull request:

- Refactor error handling in install/upgrade steps
- Implement waiting methods instead of fixed delay for delete-on-install-errror and rollback-on-upgrade-error

Image bump in #8654 

**Related issue(s)**
 #8576 #8654

### CODE REVIEW

**Comment by Tomasz-Smelcerz-SAP** [comment]
Example logs from rollback-on-upgrade-error:

//Successfull rollback:
2020/05/29 12:48:51 waiting for completion of rollback of release "dex"
2020/05/29 12:48:52 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:49:02 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:49:12 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:49:22 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:49:32 WaitForReleaseRollback() - success: Release "dex" deployed.


//Timeout during rollback:
2020/05/29 12:35:55 waiting for completion of rollback of release "dex"
2020/05/29 12:35:56 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:36:06 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:36:16 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:36:26 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:36:36 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:36:47 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:36:57 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:37:07 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:37:17 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:37:27 WaitForReleaseRollback() - pending: release found with status: kyma-system/dex: PENDING_ROLLBACK
2020/05/29 12:37:37 Timeout occured

File: components/kyma-operator/pkg/kymaoperation/steps/install.go [hunk 1]
```diff
@@ -38,36 +37,47 @@ func (s installStep) Run() error {
...
 
-			if err != nil {
-				deleteErrMsg := fmt.Sprintf("Helm delete of %s failed with an error: %s", s.component.GetReleaseName(), err.Error())
-				return errors.New(fmt.Sprintf("%s \n %s \n", installErrMsg, deleteErrMsg))
-			}
+	isDeletable, checkErr := s.helmClient.IsReleaseDeletable(s.component.GetReleaseName())
+
+	if checkErr != nil {
+		statusErrMsg := fmt.Sprintf("Checking status of release \"%s\" failed with an error: %s", s.component.GetReleaseName(), checkErr.Error())
+		log.Println(statusErrMsg)
```
Comment (colunira) [inline components/kyma-operator/pkg/kymaoperation/steps/install.go:55]: This will lead to printing the `statusErrMsg` twice

Comment (Tomasz-Smelcerz-SAP) [reply components/kyma-operator/pkg/kymaoperation/steps/install.go:55]: Done.

File: components/kyma-operator/pkg/kymahelm/wait_test.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+			//when
+			b, err := c.WaitForCondition("some", pf, ReleaseStatusFunc(rsf))
+
+			//then
+			So(err, ShouldBeNil)
+			So(b, ShouldBeTrue)
+		})
+
+		Convey("return a status on second try", func() {
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply components/kyma-operator/pkg/kymahelm/wait_test.go:38]: I think that functions that iterate internally must be tested for at least a few cases (success on first, success on second, success on n-th, no success). This increases our trust the iteration logic is correct. But better naming and explicit parameters are fine for me.

**Pull request merged**